File: app/src/main/java/org/schulcloud/mobile/data/model/Content.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmList;
import io.realm.RealmObject;

public class Content extends RealmObject{
    // text
    public String text;

    // resources
    public RealmList<Resource> resources;

    // geogebra
    public String materialId;

    // etherpad
    public String title;
    public String description;
    public String url;
}

File: app/src/main/java/org/schulcloud/mobile/models/content/Content.kt
package org.schulcloud.mobile.models.content

import io.realm.RealmList
import io.realm.RealmObject

open class Content : RealmObject() {
    // text
    var text: String? = null

    // resources
    var resources: RealmList<Resource>? = null

    // internal, etherpad, nexboard
    var url: String? = null

    // geogebra
    var materialId: String? = null

    // etherpad, nexboard
    var title: String? = null
    var description: String? = null
}
